src/app/AsteroidAppConfig.cmake.in

```diff
@@ -0,0 +1,8 @@
+@PACKAGE_INIT@
+
```

Comment:
maybe I'm missing something but I don't see how you solved the /usr/share/applications/*.desktop file generation.

If you look at src/app/asteroidapp.prf you see that all apps have their desktop file generated out of a template file:

```
desktop.commands = bash $$_PRO_FILE_PWD_/i18n/generate-desktop.sh $$_PRO_FILE_PWD_ $${TARGET}.desktop
```

And installed into /usr/share/applications:

```
desktop.files = $$OUT_PWD/$${TARGET}.desktop
desktop.path = /usr/share/applications
desktop.CONFIG = no_check_exist
```

This is a bit convoluted but this is done so that app names can be translated via the weblate UI.

Reply:
I haven't at the moment. Shouldn't be too hard to do the same, but I guess you want this done from qml-asteroid rather than on the individual applications?

Reply:
mhh idk, it feels a bit out of place in qml-asteroid as well...
the idea is to keep the maintenance overhead low for all the applications so that changing how a detail works doesn't require 30 pull requests.
But generate-desktop.sh is already provided in each app anyway so that already sucks.
If asteroidapp could provide a cmake magic trick to make it easy to generate the .desktop file, I imagine it could be nice but you know cmake better, it's your call.

Reply:
Done it in qml-asteroid for now. A CMake module is now available which provides the CMake functions `generate_desktop()` and `build_translations()`. Other CMake package can use this to generate the desktop file and build the translations. This deduplicates the `generate-desktop.sh` from every repo to just a single instance provided by qml-asteroid.

In packaging everything under `/usr/share/asteroidapp` and `/usr/bin/generate-desktop` needs splitting out to a -dev package.